scrapy/mail.py

```diff
@@ -32,8 +36,8 @@ def __init__(self, smtphost='localhost', mailfrom='scrapy@localhost',
             smtpuser=None, smtppass=None, smtpport=25, smtptls=False, smtpssl=False, debug=False):
         self.smtphost = smtphost
         self.smtpport = smtpport
-        self.smtpuser = smtpuser
-        self.smtppass = smtppass
+        self.smtpuser = to_bytes(smtpuser) if smtpuser is not None else None
```

Comment:
@redapple can you modify to_bytes() so that it accepts None, and returns None?
If you need to do "is not None else None" every place you use to_bytes, that looks ugly.

Reply:
Other uses of to_bytes() do no require this as far as I understand.
True, it may look ugly here, but I can live with that.
Let's wait for other reviews.

Reply:
Alright @rodrigc , I added a `_to_bytes_or_none` helper. Looks nicer indeed :)